astroquery/query.py

```diff
@@ -378,13 +403,21 @@ def _download_file(self, url, local_filepath, timeout=None, auth=None,
             with open(local_filepath, open_mode) as f:
                 for block in response.iter_content(blocksize):
                     f.write(block)
-                    bytes_read += blocksize
+                    bytes_read += len(block)
                     if length is not None:
                         pb.update(bytes_read if bytes_read <= length else
                                   length)
                     else:
                         pb.update(bytes_read)
-
+        if length:
+            if length > bytes_read:
+                raise RuntimeError('Received incomplete data from {}: '
+                                   '{} vs expended {}'.format(url, bytes_read,
+                                                              length))
+            elif length < bytes_read:
+                raise RuntimeError('Received extra data from {}: '
```

Comment:
if you want you can use fstrings, that's probably a slightly clearer read for these strings